### ISSUE TITLE
Refactor DI

### ISSUE THREAD

**Issue by WaveHack** [WaveHack/OpenDominion#482]
Refactor classes (including tests) to resolve dependencies out of the IoC container with `$this->dep = app($class::class)`, instead of through the constructor. Constructor arg lists get quite long (see pop calc), and needs fixing test constructors whenever this changes

---
Beep, boop, I'm [a bot](https://github.com/LeagueSandbox/IssueBot)! This issue was created by `@Sharqy` in `#bot-spam`.

**Comment by ndcroos** [comment]
Hello, I'd like to work on this issue.

**Comment by WaveHack** [comment]
@internetfett Can you check if this issue is still relevant and something @ndcroos could work on?

**Comment by internetfett** [comment]
Yes, this is still relevant in a number of places. Happy to have the help @ndcroos
